node/node.go

```diff
@@ -480,9 +481,33 @@ func (n *Node) GetAllIssuedESDTs() ([]string, error) {
 	}
 
 	for leaf := range chLeaves {
-		key := string(leaf.Key())
-		if strings.Contains(key, "-") {
-			tokens = append(tokens, key)
+		tokenName := string(leaf.Key())
+		if !strings.Contains(tokenName, "-") {
+			continue
+		}
+
+		if tokenType == "" {
+			tokens = append(tokens, tokenName)
+			continue
+		}
+
+		esdtToken := &systemSmartContracts.ESDTData{}
+		suffix := append(leaf.Key(), userAccount.AddressBytes()...)
+		value, errVal := leaf.ValueWithoutSuffix(suffix)
+		if errVal != nil {
+			log.Warn("cannot get value without suffix", "error", errVal, "key", leaf.Key())
+			continue
+		}
+
+		err = n.internalMarshalizer.Unmarshal(esdtToken, value)
+		if err != nil {
+			log.Warn("cannot unmarshal", "token name", tokenName, "err", err)
+			continue
+		}
+
+		expectedTokenType := []byte(tokenType)
```

Comment:
you don't need an extra variable here
---> to speed up the execution you can remove this extra variable

Reply:
done